components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessor.java

```diff
@@ -87,6 +88,8 @@ public InboundProcessorResponseDTO handleRequest(int msgSize, String msgText,
                             // subscription operation name
                             String subscriptionOperation = GraphQLProcessorUtil.getOperationList(operation,
                                     inboundMessageContext.getGraphQLSchemaDTO().getTypeDefinitionRegistry());
+                            WebSocketUtils.setApiPropertyToChannel(inboundMessageContext.getCtx(),
```

Comment:
Dont we need to add the same fix to GraphQLResponseProcessor in handleResponse()? If someone use the same connection for two different subscriptions what would be the behavior?

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/11166/commits/fff020f1e84428a67f6601c736bbcbc1b2ca6cf9. Tested with multiple subscriptions in the same connections and shown below is show it looks in portal, correct resource path is mentioned in the table.
<img width="1381" alt="Screenshot 2022-03-18 at 06 35 00" src="https://user-images.githubusercontent.com/28379317/158937573-43f8e2fb-3fe6-41ac-97c9-c32ba391212a.png">